File: backend/app/api/v1/schemas/device.py
from pydantic import BaseModel
from typing import List, Optional

class DeviceLocation(BaseModel):
    city: str
    country: str
    facility: Optional[str] = None
    coordinates: Optional[List[float]] = None

class DeviceNetwork(BaseModel):
    asn: int
    as_name: str

class DeviceResponse(BaseModel):
    id: str
    name: str
    platform: str
    location: DeviceLocation
    supported_queries: List[str]
    status: str = "online"
    network: DeviceNetwork

class DeviceListResponse(BaseModel):
    devices: List[DeviceResponse]

File: frontend/src/types/device.ts
export interface DeviceLocation {
  city: string;
  country: string;
  facility?: string;
  coordinates?: [number, number];
}

export interface DeviceNetwork {
  asn: number;
  as_name: string;
}

export interface Device {
  id: string;
  name: string;
  platform: string;
  location: DeviceLocation;
  supported_queries: string[];
  status: string;
  network: DeviceNetwork;
}

export interface DeviceListResponse {
  devices: Device[];
}
